leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java

```diff
@@ -190,35 +221,89 @@ public static void createAndStartServer(int webPort, String localAddress, int lo
-            byte[] publicY = Hex
-                    .decodeHex("d2ffaa73957d76984633fc1cc54d0b763ca0559a9dff9706e9f4557dacc3f52a".toCharArray());
-            byte[] privateS = Hex
-                    .decodeHex("1dae121ba406802ef07c193c1ee4df91115aabd79c1ed7f4c0ef7ef6a5449400".toCharArray());
-
-            // Get Elliptic Curve Parameter spec for secp256r1
-            AlgorithmParameters algoParameters = AlgorithmParameters.getInstance("EC");
-            algoParameters.init(new ECGenParameterSpec("secp256r1"));
-            ECParameterSpec parameterSpec = algoParameters.getParameterSpec(ECParameterSpec.class);
-
-            // Create key specs
-            KeySpec publicKeySpec = new ECPublicKeySpec(new ECPoint(new BigInteger(publicX), new BigInteger(publicY)),
-                    parameterSpec);
-            KeySpec privateKeySpec = new ECPrivateKeySpec(new BigInteger(privateS), parameterSpec);
-
-            // Get keys
-            publicKey = KeyFactory.getInstance("EC").generatePublic(publicKeySpec);
-            PrivateKey privateKey = KeyFactory.getInstance("EC").generatePrivate(privateKeySpec);
-            builder.setPublicKey(publicKey);
-            builder.setPrivateKey(privateKey);
-        } catch (InvalidKeySpecException | NoSuchAlgorithmException | InvalidParameterSpecException e) {
-            LOG.error("Unable to initialize RPK.", e);
-            System.exit(-1);
+
+        // Set up X.509 mode
+        if (keyStorePath != null) {
+            try {
+                KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+                try (FileInputStream fis = new FileInputStream(keyStorePath)) {
+                    keyStore.load(fis, keyStorePass == null ? null : keyStorePass.toCharArray());
+                    ArrayList<Certificate> trustedCertificates = new ArrayList<>();
+                    for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();) {
+                        String alias = aliases.nextElement();
+                        if (keyStore.isCertificateEntry(alias)) {
+                            trustedCertificates.add(keyStore.getCertificate(alias));
+                        } else if (keyStore.isKeyEntry(alias) && alias.equals(keyStoreAlias)) {
+                            ArrayList<X509Certificate> x509CertificateChain = new ArrayList<>();
+                            Certificate[] certificateChain = keyStore.getCertificateChain(alias);
+                            if (certificateChain != null) {
+                                for (Certificate certificate : certificateChain) {
+                                    if (certificate instanceof X509Certificate) {
+                                        x509CertificateChain.add((X509Certificate) certificate);
+                                    } else {
+                                        LOG.warn("Ignoring non-X.509 certificate in alias chain: {}", certificate);
```

Comment:
this is probably an error if you have a non-X509 cert in the cert-chain, so an exception or any fatal error is better